### DISCUSSION TITLE
Provide a way to document Astro components, integrated with the language server

### DISCUSSION THREAD

**Discussion by JonasGruenwald** [withastro/roadmap#966] [Proposal]
# Summary

It would be nice to have a standard way of documenting Astro components which the language server understands, so component documentation can be shown on hover for a given component.

# Background & Motivation

When creating re-usable component within a project or components for use by others, it is useful to document the components functionality and props for reference.

Most frameworks provide ways to do this:

* In Svelte you can use HTML comments with the `@component` tag for this: https://github.com/sveltejs/language-tools/pull/285
* In React, since components are just functions (or classes), documentation can be easily attached using existing standard jsdoc methods

# Goals

- Provide a standard way of documenting Astro components
- Adjust the language server accordingly, so that it can show component documentation on hover

**Comment by JonasGruenwald** [comment]
Ah I suppose I missed this:
https://github.com/withastro/roadmap/blob/main/proposals/0026-component-interface.md

I'm closing this since the linked proposal was accepted

**Reply by markomitranic** [reply]
Accepted in 2022 but never acted upon ;) i hope they implement it soon, this is a no brainer

**Reply by lucasfsduarte** [reply]
Any updates here? Early 2026 and seems that this wasn't acted upon yet.
